node_modules/oae-principals/tests/test-email.js

```diff
@@ -0,0 +1,280 @@
+
+                        // Assert that the email contains both plain text and html
+                        assert.ok(message.html);
+                        assert.ok(message.text);
+
+                        // Assert a URL is included that can be used to verify the email address
+                        assert.ok(message.html.match(/\?token=[a-zA-Z0-9]{32}/));
+                        assert.ok(message.text.match(/\?token=[a-zA-Z0-9]{32}/));
+
+                        // Assert the email is not verified yet
+                        RestAPI.User.getMe(simong.restContext, function(err, me) {
+                            assert.ok(!err);
+                            assert.strictEqual(me.email, email);
+                            assert.strictEqual(me.emailVerified, false);
+
+                            return callback();
+                        });
+                    });
+                });
+            });
+        });
+
+        describe('Verifying email tokens', function() {
+
+            /**
+             * Test that verifies that an email token can be used to verify an email address
+             */
+            it('verify an email token can be used to verify an email address', function(callback) {
+                TestsUtil.generateTestUsers(camAdminRestContext, 1, function(err, users, simong) {
+                    assert.ok(!err);
+
+                    var email = TestsUtil.generateTestEmailAddress();
+                    RestAPI.User.updateUser(simong.restContext, simong.user.id, {'email': email}, function(err) {
+                        assert.ok(!err);
+                    });
+
+                    EmailAPI.once('debugSent', function(message) {
+                        assert.ok(message.text);
+
+                        // Get the token from the email
+                        var token = message.text.match(/\?token=([a-zA-Z0-9]{32})/)[1];
+                        assert.ok(token);
+
+                        // Verify the token
+                        RestAPI.User.verifyEmailToken(simong.restContext, token, function(err) {
+                            assert.ok(!err);
+
+                            // Assert that the email was verified
+                            RestAPI.User.getMe(simong.restContext, function(err, me) {
+                                assert.ok(!err);
+                                assert.strictEqual(me.emailVerified, true);
+
+                                return callback();
+                            });
+                        });
+                    });
+                });
+            });
+
+            /**
+             * Test that verifies the email tokens are validated
+             */
+            it('verify validation', function(callback) {
+                TestsUtil.generateTestUsers(camAdminRestContext, 1, function(err, users, simong) {
+                    assert.ok(!err);
+
+                    RestAPI.User.verifyEmailToken(simong.restContext, null, function(err) {
```

Comment:
Do you have to be signed in to validate an email token? If not, it might be worth doing this as an anonymous user. If you have to be signed in, it might be worth asserting that you cannot validate a token from a different user.